frontend/src/pages/projects/screens/detail/types.ts

```diff
@@ -4,6 +4,15 @@ export enum ProjectSectionID {
   DATA_CONNECTIONS = 'data-connections',
 }
 
+// Section behind feature flag
+export enum ProjectSectionIDExtended {
+  MODEL_SERVER = 'model-server',
+}
+
 export type ProjectSectionTitlesType = {
   [key in ProjectSectionID]: string;
 };
+
+export type ProjectSectionTitlesTypeExtended = ProjectSectionTitlesType & {
```

Comment:
🤔 This `...Extended` naming is weird, it explains nothing of what it actually is. I am wondering if `required` vs `optional` is better -- but that could be equally weird. Maybe we just name it by the feature? 🤔 

What are your thoughts, when we add a new section that is also feature flagged, our model should be easily extendable at that point.

Note this is all about organization, I don't see anything inherently wrong with what you did. Almost might be worth to just make the enum values optional in `ProjectSectionTitlesType` and put everything together.